decision-records/2024-06-04--service-to-stop-while-waiting-for-cert.md

```diff
@@ -0,0 +1,106 @@
+**Date:** 2024-06-04<br/>
+**Author:** @sed-i
+
+## Context and Problem Statement
+When an admin forms a tls relation, there is a time window in which:
+
+1. the juju model tells us we must operate with TLS enabled; however
+2. we haven't received the server cert yet.
+
+As a result we had some code ordering issues (e.g. #414):
+- Should we update the config file with the tls config, which includes path to
+  the cert, before we got the cert, only to perfectly reflect the juju model?
+- Should we wait with the tls config and allow the workload to continue to
+  operate in plain http, in contradiction to the juju model?
+- How should we determine within the charm if "TLS is enabled"?
+
+This problem is common to many charms.
+
+## Decision
+A server shall NOT keep using plain http while waiting for a cert. The service
```

Comment:
I'd phrase this more generally, it's not just about waiting for the cert, it's about the cert being on disk (workload AND charm): what if pebble is temporarily offline or there's a glitch in the matrix and secret-get returns rubbish?
In general, I'd say, "the charm shall NOT keep using plain http during any of the transient intermediate states"

Reply:
Updated.